### ISSUE TITLE
Doesn't work on Arm/Arm64 devices.

### ISSUE THREAD

**Issue by Cee123** [libretro/melonDS#62]
Hey there, I really like this emulator. It is a fantastic alternative to Desmume and works excellently. However, when I try to get it working on Rapberry Pi 3B, Odroid XU4 or Odroid N2, it just crashes. Seems to be something wrong with the version for Arm Linux/Android devices.

**Comment by bvkuz** [comment]
Yes, same here my screen goes black and my rpi restarts.

**Comment by Cee123** [comment]
> Yes, same here my screen goes black and my rpi restarts.

Yeah same. Wont work on any of my single board computers. Wont even work on an android device. Would be great to be able to use this as an alternative to drastic.

**Comment by bvkuz** [comment]
I've managed to make it work. I've crated two folders on the resources folder in the 
addon_data\game.libretro.melonds_libretro_buildbot\resources\
one was system where I placed my bios files
and settings where I placed my settings.xml

And it's working but unfortunately i'ts very laggy.

**Comment by Cee123** [comment]
> I've managed to make it work. I've crated two folders on the resources folder in the
> addon_data\game.libretro.melonds_libretro_buildbot\resources
> one was system where I placed my bios files
> and settings where I placed my settings.xml
> 
> And it's working but unfortunately i'ts very laggy.

Awesome! That's fantastic! Yeah unfortunately i dont think there'll be much better performance than with desmume. I wonder if the speed would be playable on something powerful like the odroid n2 plus.

**Comment by Cee123** [comment]
> I've managed to make it work. I've crated two folders on the resources folder in the
> addon_data\game.libretro.melonds_libretro_buildbot\resources
> one was system where I placed my bios files
> and settings where I placed my settings.xml
> 
> And it's working but unfortunately i'ts very laggy.

That resources folder, is that on Retropie? I wonder if Emuelec has a resources folder where i can add those files.

**Comment by Cee123** [comment]
> I've managed to make it work. I've crated two folders on the resources folder in the
> addon_data\game.libretro.melonds_libretro_buildbot\resources
> one was system where I placed my bios files
> and settings where I placed my settings.xml
> 
> And it's working but unfortunately i'ts very laggy.

Just realized you were talking about getting Melonds working on Kodi. Im talking about getting it working on things like RetroPie, RetroArena/Ares or EmuElec.

**Comment by Cee123** [comment]
My issue seems to be related to this in the log file
> Error(s): /opt/ares/libretrocores/lr-melonds/melonds_libretro.so: wrong ELF class: ELFCLASS32

**Comment by Cee123** [comment]
Update: I finally got it working on RetroPie, RetroArena and EmuElec. What I was doing wrong was I was trying to use a core that was from the nightly retroarch builds. So what I did was I compiled my own versions using the devices I wanted to use them on and used the ones I compiled. It works!!

BUT it's slow and laggy. Even on the powerful Odroid N2 Plus, I was expecting this to be smooth and playable at full speed but it wasn't. Think I'll stick to Drastic.
